addons/website/tools.py

```diff
@@ -1,188 +1,8 @@
-        ),
-        lang=env['res.lang']._lang_get(lang_code),
-        redirect=env['ir.http']._redirect,
-        session=DotDict(
-            geoip={'country_code': country_code},
-            debug=False,
-            sale_order_id=sale_order_id,
-            website_sale_current_pl=website_sale_current_pl,
-        ),
-        website=website
-    )
-
-    with contextlib.ExitStack() as s:
-        odoo.http._request_stack.push(request)
-        s.callback(odoo.http._request_stack.pop)
-        s.enter_context(patch('odoo.http.root.get_db_router', router))
-
-        yield request
-
-# Fuzzy matching tools
-
-def distance(s1="", s2="", limit=4):
-    """
-    Limited Levenshtein-ish distance (inspired from Apache text common)
-    Note: this does not return quick results for simple cases (empty string, equal strings)
-        those checks should be done outside loops that use this function.
-
-    :param s1: first string
-    :param s2: second string
-    :param limit: maximum distance to take into account, return -1 if exceeded
-
-    :return: number of character changes needed to transform s1 into s2 or -1 if this exceeds the limit
-    """
-    BIG = 100000 # never reached integer
-    if len(s1) > len(s2):
-        s1, s2 = s2, s1
-    l1 = len(s1)
-    l2 = len(s2)
-    if l2 - l1 > limit:
-        return -1
-    boundary = min(l1, limit) + 1
-    p = [i if i < boundary else BIG for i in range(0, l1 + 1)]
-    d = [BIG for _ in range(0, l1 + 1)]
-    for j in range(1, l2 + 1):
-        j2 = s2[j -1]
-        d[0] = j
-        range_min = max(1, j - limit)
-        range_max = min(l1, j + limit)
-        if range_min > 1:
-            d[range_min -1] = BIG
-        for i in range(range_min, range_max + 1):
-            if s1[i - 1] == j2:
-                d[i] = p[i - 1]
-            else:
-                d[i] = 1 + min(d[i - 1], p[i], p[i - 1])
-        p, d = d, p
-    return p[l1] if p[l1] <= limit else -1
-
-def similarity_score(s1, s2):
-    """
-    Computes a score that describes how much two strings are matching.
-
-    :param s1: first string
-    :param s2: second string
-
-    :return: float score, the higher the more similar
-        pairs returning non-positive scores should be considered non similar
-    """
-    dist = distance(s1, s2)
-    if dist == -1:
-        return -1
-    set1 = set(s1)
-    score = len(set1.intersection(s2)) / len(set1)
-    score -= dist / len(s1)
-    score -= len(set1.symmetric_difference(s2)) / (len(s1) + len(s2))
-    return score
-
-def text_from_html(html_fragment):
-    """
-    Returns the plain non-tag text from an html
-
-    :param html_fragment: document from which text must be extracted
-
-    :return: text extracted from the html
-    """
-    # lxml requires one single root element
-    tree = etree.fromstring('<p>%s</p>' % html_fragment, etree.XMLParser(recover=True))
-    return ' '.join(tree.itertext())
+# Backwards compatibility
+from odoo.tests.common import MockRequest
```

Comment:
This one at least can be removed imo, retro-compat for tests is probably overkill (if it was used outside of tests by a custo, it's weird and not worth the retro compat).
Others can also probably be removed, I am not sure we usually keep retro-compat stuff like that when moving utils method?

Reply:
It's just to avoid failing all tests that rely on importing `MockRequests` from this file.

Odoo maintains retro-compatibility in absurdly old coding patterns like below, so I thought it'd be a good idea for you guys:

https://github.com/odoo/odoo/blob/edc88a2ab99673427cbeac05246b8e586b121b72/odoo/osv/osv.py#L7-L13

In any case, I don't have write access here anymore, and it has conflicts, so you will have to fix those yourselves when merging... sorry about that 😕

Reply:
Tell me which change to do, and I'll do it.

Reply:
I was asking to remove the `MockRequest` retro-compatibility which seemed overkill as it's test oriented.
I'd not have kept the other ones either.

But if it's annoying to do, because it will required to adapt all imports, I am fine as is.
Anyway, I won't be the one to r+ this one as it's moving utils away from website to base, it's out of our scope, so just probably wait to see if @xmo-odoo  wants to drop them too or not.